File: csharp/darts/Darts.cs
using System;

namespace Exercism.CSharp.Solutions.DartsExercise
{
    public static class Darts
    {
        private struct Points
        {
            public const int Outside        = 0;
            public const int OuterCircle   = 1;
            public const int MiddleCircle  = 5;
            public const int InnerCircle   = 10;
        }

        private struct Radius
        {
            public const double OuterCircle    = 10.0d;
            public const double MiddleCircle   = 5.0d;
            public const double InnerCircle    = 1.0d;
        }

        public static int Score(double x, double y)
        {
            double landingRadius = GetRadius(x, y);

            if (landingRadius <= Radius.InnerCircle)
            {
                return Points.InnerCircle;
            }
            else if (landingRadius <= Radius.MiddleCircle)
            {
                return Points.MiddleCircle;
            }
            else if (landingRadius <= Radius.OuterCircle)
            {
                return Points.OuterCircle;
            }

            return Points.Outside;
        }

        private static double GetRadius(double x, double y) =>
            Math.Sqrt((x * x) + (y * y));
    }
}

File: c/darts/src/darts.c
#include <math.h>
#include "darts.h"

#define SCORE_OUTSIDE           0
#define SCORE_OUTER_CIRCLE      1
#define SCORE_MIDDLE_CIRCLE     5
#define SCORE_INNER_CIRCLE      10

#define RADIUS_OUTER_CIRCLE     10.0F
#define RADIUS_MIDDLE_CIRCLE    5.0F
#define RADIUS_INNER_CIRCLE     1.0F

static float get_radius(coordinate_t landing_position);

uint8_t score(coordinate_t landing_position)
{
    float landing_radius = get_radius(landing_position);

    if (landing_radius <= RADIUS_INNER_CIRCLE)
    {
        return SCORE_INNER_CIRCLE;
    }
    else if (landing_radius <= RADIUS_MIDDLE_CIRCLE)
    {
        return SCORE_MIDDLE_CIRCLE;
    }
    else if (landing_radius <= RADIUS_OUTER_CIRCLE)
    {
        return SCORE_OUTER_CIRCLE;
    }

    return SCORE_OUTSIDE;
}

static float get_radius(coordinate_t landing_position)
{
    return sqrtf(powf(landing_position.x, 2.0f) + powf(landing_position.y, 2.0f));
}
